sentry/src/transport.rs

```diff
@@ -376,15 +402,130 @@ implement_http_transport! {
     }
 }
 
+#[cfg(feature = "with_surf_transport")]
+implement_http_transport! {
+    /// A transport can send events via HTTP to sentry via `surf`.
+    ///
+    /// This is enabled by the `with_surf_transport` flag.
+    pub struct SurfHttpTransport;
+
+    fn spawn(
+        options: &ClientOptions,
+        receiver: Receiver<Option<Event<'static>>>,
+        signal: Arc<Condvar>,
+        shutdown_immediately: Arc<AtomicBool>,
+        queue_size: Arc<Mutex<usize>>,
+        http_client: SurfClient<NativeClient>,
+    ) {
+        let dsn = options.dsn.clone().unwrap();
+        let user_agent = options.user_agent.to_string();
+        let mut disabled = None::<SystemTime>;
+
+        thread::Builder::new()
+            .name("sentry-transport".to_string())
+            .spawn(move || {
+                sentry_debug!("spawning surf transport");
+                let http_client = http_client;
+                let url = dsn.store_api_url().to_string();
```

Comment:
You are doing a `to_string()` here, and a `as_str()` later on. But I see the other transports do that as well.